kernel/src/fs/procfs/pid/status.rs

```diff
@@ -59,54 +60,87 @@ use crate::{
 /// - Mems_allowed_list: List of memory nodes allowed for this process.
 /// - voluntary_ctxt_switches: Number of voluntary context switches.
 /// - nonvoluntary_ctxt_switches: Number of nonvoluntary context switches.
-pub struct StatusFileOps(Arc<Process>);
+pub struct StatusFileOps {
+    process_ref: Arc<Process>,
+    thread_ref: Arc<Thread>,
+}
 
 impl StatusFileOps {
-    pub fn new_inode(process_ref: Arc<Process>, parent: Weak<dyn Inode>) -> Arc<dyn Inode> {
-        ProcFileBuilder::new(Self(process_ref))
-            .parent(parent)
-            .build()
-            .unwrap()
+    pub fn new_inode(
+        process_ref: Arc<Process>,
+        thread_ref: Arc<Thread>,
+        parent: Weak<dyn Inode>,
+    ) -> Arc<dyn Inode> {
+        ProcFileBuilder::new(Self {
+            process_ref,
+            thread_ref,
+        })
+        .parent(parent)
+        .build()
+        .unwrap()
     }
 }
 
 impl FileOps for StatusFileOps {
     fn data(&self) -> Result<Vec<u8>> {
-        let process = &self.0;
-        let main_thread = process.main_thread();
-        let file_table = main_thread.as_posix_thread().unwrap().file_table();
+        let process = &self.process_ref;
+        let thread = &self.thread_ref;
+        let posix_thread = thread.as_posix_thread().unwrap();
 
         let mut status_output = String::new();
-        writeln!(status_output, "Name:\t{}", process.executable_path()).unwrap();
+        writeln!(
+            status_output,
+            "Name:\t{}",
+            posix_thread
+                .thread_name()
+                .lock()
+                .as_ref()
+                .and_then(|name| name.as_string())
+                .unwrap_or_else(|| process.executable_path())
+        )
+        .unwrap();
+
+        let state = if thread.is_exited() {
+            "Z (zombie)"
+        } else {
+            "R (running)"
+        };
+        writeln!(status_output, "State:\t{}", state).unwrap();
+
         writeln!(status_output, "Tgid:\t{}", process.pid()).unwrap();
-        writeln!(status_output, "Pid:\t{}", process.pid()).unwrap();
+        writeln!(status_output, "Pid:\t{}", posix_thread.tid()).unwrap();
         writeln!(status_output, "PPid:\t{}", process.parent().pid()).unwrap();
         writeln!(status_output, "TracerPid:\t{}", process.parent().pid()).unwrap(); // Assuming TracerPid is the same as PPid
```

Comment:
Well, it's not this PR's fault. We don't even have `ptrace` support, so why should we use the parent PID as the tracer PID? I think it should be zero to indicate that there are no tracers.